File: Valghalla.Internal.Application/Modules/Tasks/Responses/TaskStatusGeneralInfoResponse.cs
namespace Valghalla.Internal.Application.Modules.Tasks.Responses
{
    public sealed record TaskStatusGeneralInfoResponse
    {       
        public int AssignedTasksCount { get; init; }
        public int MissingTasksCount { get; init; }
        public int AwaitingTasksCount { get; init; }
        public int AllTasksCount { get; init; }
        public int RejectedTasksCount { get; set; }
        public List<RejectedTasksInfoResponse> RejectedTasksInfoResponses { get; set; } = new List<RejectedTasksInfoResponse>();
    }
}

File: Valghalla.Internal.Web/src/features/tasks/models/task-status-general-info-response.ts
import { RejectedTasksInfoResponse } from './rejected-tasks-info-response';

export interface TaskStatusGeneralInfoResponse {
  assignedTasksCount: number;
  missingTasksCount: number;
  awaitingTasksCount: number;
  allTasksCount: number;
  rejectedTasksCount: number;
  rejectedTasksInfoResponses: RejectedTasksInfoResponse[];
}
